docs/manuals/gui/extension.md

```diff
@@ -1,42 +1,137 @@
+the element as it appears in pages, in the `(ElementLibrary.)get_elements()^` method.
 
-- The Javascript part.<br/>
-  Visual element declarations indicate how the browser should display the element
-  and what interactions are available.
+The [`Element` constructor](Element.__init__()^) needs a description of all the properties
+that this element holds, as well as how this element is rendered.
+
+```py
+Element("<default_property_name>",
+       {
+           "<property_1_name>": ElementProperty(<property_1_type>, ...),
+           "<property_2_name>": ElementProperty(<property_2_type>, ...),
+           ...
+        },
+        <rendering_arguments>)
+```
+
+An element property descriptor (handled by the `ElementProperty^` class) must be
+indicated its type (one of the `PropertyType^` values) and potentially a default
+value.
+
+All elements must indicated what is its default property name. That is used in Markdown
+page as the first fragment of the `<|...|>` construct, or the value located in the
+text part of an element tag in HTML pages.
+
+Of course *default_property_name* must be one of the keys of the properties dictionary
+provided to the [`Element` constructor](Element.__init__()^).
+
+## Rendering Elements
+
+The Taipy GUI Extension package provides two different ways to implement the rendering
+of the element and its interactions. Both approaches deliver HTML fragments that are
+inserted in the page when it is requested.
+
+- Static elements.<br/>
+  A static visual element is one that cannot interact with the underlying application.
+  It can be used just like any other element in a page, but its properties are not
+  bound to the application variable: if the variable value is modified, it does not
+  modify the representation of the element on the page.<br/>
+  Static elements are implemented by creating a string that holds the XHTML text (that is,
+  HTML that respects the XML syntax, where all tags must be closed) that is inserted in
+  the page displayed by the browser.<br/>
+  Please go to the [Static Elements Example](extension_static_element.md) page for
+  a complete description of how to implement your own static custom elements.
+
+- Dynamic elements.<br/>
+  Dynamic elements provide the binding functionality of Taipy GUI: if a property value
+  depends on an application variable (and if the property type is *dynamic*) then the
+  page automatically updates when the variable value changes.<br/>
+  In Taipy GUI, dynamic visual element are implemented using the
+  [React](https://reactjs.org/) JavaScript library, and the
+  [TypeScript](https://www.typescriptlang.org/) programming language (that builds on
+  JavaScript).
 
 ## Prerequisites
 
 In order to create and use custom visual elements, you need to install:
 
-- Python 3.8 or higher.
 - Taipy GUI 2.0 or higher (included in Taipy and Taipy Enterprise).
-- NodeJS
-- NPM
+- Python 3.8 or higher.
+- NodeJS and NPM if you plan to create dynamic custom visual elements.
```

Comment:
Npm ?

Reply:
ok ok I will make that slighlty more discrete.